### ISSUE TITLE
New InputBarAccessoryView

### ISSUE THREAD

**Issue by nathantannar4** [MessageKit/MessageKit#83]
A new re-written accessory view to be used to create robust and customizable MessageInputBar's

**Comment by hyouuu** [comment]
1 minor issue I found with the current inputAccessoryView is that, if you return too many times that the textView expands above the screen top, you cannot scroll to the top of the message ;)

**Comment by nathantannar4** [comment]
Yes that is not in the new one I made. There is a maxHeight variable that can be used to correct that

**Comment by hyouuu** [comment]
Cool - or make it scrollable if beyond screen?

**Comment by nathantannar4** [comment]
Once it reaches the maxHeight the textview becomes scrollable. Same way the iMessage app works

**Comment by etoledom** [comment]
When the textView grows and reaches maxHeight, after sending the message it doesn't go back to the original size.

**Comment by nathantannar4** [comment]
Good catch @etoledom . I have isolated the bug to changing the `UITextView.isScrollEnabled: Bool` property. For some reason it doesn't let the view resize, possibly due to the auto layout constraints? I am not sure.

This can be corrected by forcing a temp heightAnchor on the InputTextView.


Set the property in setupConstraints(): `textViewLayoutSet?.height = textView.heightAnchor.constraint(equalToConstant: 36)`
And modify textViewDidChange() to include: `textViewLayoutSet?.height?.isActive = textView.text.isEmpty`

**Comment by nathantannar4** [comment]
Never mind that has bugs, call `textView.invalidateIntrinsicContentSize()` in
```Swift
override open var intrinsicContentSize: CGSize {
        let sizeToFit = textView.sizeThatFits(CGSize(width: textView.bounds.width, height: .greatestFiniteMagnitude))
        var heightToFit = sizeToFit.height.rounded() + padding.top + padding.bottom + separatorLine.frame.height + tableViewHeightConstant
        if heightToFit >= maxHeight {
            textView.isScrollEnabled = true
            heightToFit = maxHeight
        } else {
            textView.isScrollEnabled = false
            textView.invalidateIntrinsicContentSize()
        }
        let size = CGSize(width: bounds.width, height: heightToFit)
        if previousIntrinsicContentSize != size {
            delegate?.inputBar(self, didChangeIntrinsicContentTo: size)
        }
        previousIntrinsicContentSize = size
        return size
    }
```

**Comment by SD10** [comment]
Closing this as it is being implemented in #80 👍 Great work @nathantannar4 🎉
